core/src/main/java/tech/tablesaw/table/Rows.java

```diff
@@ -14,36 +14,40 @@
 
 package tech.tablesaw.table;
 
-import it.unimi.dsi.fastutil.ints.IntArrayList;
+import javax.annotation.concurrent.Immutable;
+
 import tech.tablesaw.api.ColumnType;
 import tech.tablesaw.api.Table;
+import tech.tablesaw.columns.Column;
+import tech.tablesaw.selection.BitmapBackedSelection;
 import tech.tablesaw.selection.Selection;
 
-import javax.annotation.concurrent.Immutable;
-
 /**
  * A static utility class for row operations
  */
 @Immutable
-public class Rows {
+public final class Rows {
 
     // Don't instantiate
     private Rows() {}
 
-    public static void copyRowsToTable(IntArrayList rows, Table oldTable, Table newTable) {
-        for (int i = 0; i < rows.size(); i++) {
-            int rowIndex = rows.getInt(i);
-            int columnCount = oldTable.columnCount();
-            for (int columnIndex = 0; columnIndex < columnCount; columnIndex++) {
-                newTable.column(columnIndex).appendObj(oldTable.column(columnIndex).get(rowIndex));
-            }
+    /**
+     * N.B. this method does not preserve ordering. Use the version taking int[] if row ordering matters 
+     */
+    public static void copyRowsToTable(Selection rows, Table oldTable, Table newTable) {
+        copyRowsToTable(rows.toArray(), oldTable, newTable);
+    }
+
+    @SuppressWarnings({"rawtypes","unchecked"})
+    public static void copyRowsToTable(int[] rows, Table oldTable, Table newTable) {
+        for (int columnIndex = 0; columnIndex < oldTable.columnCount(); columnIndex++) {
+            Column oldColumn = oldTable.column(columnIndex);
+            newTable.column(columnIndex).append(oldColumn.subset(rows));
```

Comment:
This is much better, but i wonder if there's a way to do it without creating the intermediate column in oldColumn.subset(rows). Not a big deal, but the append method that takes a column and a row might be used here by iterating on the array, no? 

Also, maybe the same approach would work on the selection version of copyRowsToTable(), avoiding both the conversion to an int[] and the subsequent column creation step in subset(). IDK, it's really early here, so....

Reply:
That's true. That would save memory

The append method taking a column and a row would call `Preconditions.checkArgument(column.type() == this.type());` on ever cell, which might take a bit more CPU. I'm not sure if that's an issue at all?